File: go/day2.go
package main

import (
	"strconv"
	"strings"
)

func day2() {
	a1 := 0

	for _, line := range input {
		max := 0
		min := 1000000000
		for _, word := range strings.Split(line, "\t") {
			v, _ := strconv.Atoi(word)
			if v > max {
				max = v
			}
			if v < min {
				min = v
			}
		}
		a1 += max - min
	}

	a2 := 0

	for _, line := range input {
		var ints []int
		for _, word := range strings.Split(line, "\t") {
			v, _ := strconv.Atoi(word)
			ints = append(ints, v)
		}
		for _, i := range ints {
			for _, j := range ints {
				if i > j && i%j == 0 {
					a2 += i / j
				}
			}
		}
	}

	println(a1, a2)
}

File: ruby/day2.rb
input = File.open('input/input02').read.chomp.lines
a1 = 0
a2 = 0

input.each do |line|
  vals = line.split.map(&:to_i)
  a1 += vals.max - vals.min

  vals.each do |i|
    vals.each do |j|
      a2 += (i / j) if i > j && (i % j).zero?
    end
  end
end

puts a1, a2
